### PR TITLE
Feat/trn xrpl token bridge

### PR DESCRIPTION
**Pull request by JasonTulp**
Adds support for bridging tokens other than XRP from TRN -> XRPL
Adds a new withdraw extrinsic which enables withdrawing any token from TRN (Including XRP) 
In future we should remove the existing `withdraw_xrp` and `withdraw_xrp_with_destination_flag` extrinsics to tidy things up
[issue created for the above](https://futureverse.atlassian.net/jira/software/c/projects/TRN/boards/39/backlog?selectedIssue=TRN-514)

### CODE REVIEW

File: pallet/xrpl-bridge/src/lib.rs [hunk 1]
```diff
@@ -1076,33 +1197,108 @@ impl<T: Config> Pallet<T> {
...
+		destination: XrplAccountId,
+		amount: Balance,
+		tx_fee: u64,
+		who: AccountOf<T>,
+	) -> Result<WithdrawTransaction, DispatchError> {
+		let xrpl_currency =
+			AssetIdToXRPL::<T>::get(asset_id).ok_or(Error::<T>::AssetNotSupported)?;
+		// the door address pays the tx fee on XRPL. Therefore we must charge the user the tx fee
+		// in XRP alongside their asset withdrawal
+		let _ = T::MultiCurrency::burn_from(
```
Comment (KarishmaBothara) [inline pallet/xrpl-bridge/src/lib.rs:1265]: Liked the way you have structured the code... Do we want to move the burn_from part from all the 3 ways of withdrawal in add_to_withdraw ? as it is common code

File: pallet/xrpl-bridge/src/lib.rs [hunk 2]
```diff
@@ -1054,17 +1134,58 @@ impl<T: Config> Pallet<T> {
...
+			a if a == T::NativeAssetId::get() => {
+				Self::process_root_withdrawal(destination, amount, tx_fee, who.clone())?
+			},
+			_ => {
+				Self::process_asset_withdrawal(asset_id, destination, amount, tx_fee, who.clone())?
+			},
+		};
+
+		// Check if there is a payment delay and delay the payment if necessary
+		if let Some((payment_threshold, delay)) = PaymentDelay::<T>::get() {
```
Comment (KarishmaBothara) [inline pallet/xrpl-bridge/src/lib.rs:1164]: Not for now... may be for later https://github.com/futureversecom/trn-seed/blob/09961c9243098c10ad65363534861a824ce74d8c/pallet/xrpl-bridge/src/lib.rs#L296
here same payment_threshold is checked against all the tokens XRP, ASTO, SYLO, ZRP etc... may be later we might want to have it per asset id?

File: pallet/xrpl-bridge/src/lib.rs [hunk 3]
```diff
@@ -1117,14 +1313,19 @@ impl<T: Config> Pallet<T> {
 			.map_err(|_| Error::<T>::DelayScheduleAtCapacity)?;
 		DelayedPayments::<T>::insert(
 			delayed_payment_id,
-			DelayedWithdrawal { sender: sender.clone(), destination_tag, withdraw_tx: withdrawal },
+			DelayedWithdrawal {
```
Comment (KarishmaBothara) [inline pallet/xrpl-bridge/src/lib.rs:1316]: Do we also want to include asset id? we can do it in another PR... but now storage should also reflect the asset id for which delay payment happens.

File: pallet/xrpl-bridge/src/lib.rs [hunk 4]
```diff
@@ -38,20 +39,24 @@ use seed_primitives::{
...
 use sp_std::{prelude::*, vec};
 use xrpl_codec::{
+	field::Amount,
 	traits::BinarySerialize,
 	transaction::{Payment, PaymentWithDestinationTag, SignerListSet},
 };
+use xrpl_codec::{
+	transaction::PaymentAltCurrency,
+	types::{AccountIdType, AmountType, IssuedAmountType, IssuedValueType},
+};
```
Comment (surangap) [inline pallet/xrpl-bridge/src/lib.rs:55]: add to above

File: pallet/xrpl-bridge/src/lib.rs (hunk 1 above)
Comment (JasonTulp) [reply pallet/xrpl-bridge/src/lib.rs:1265]: This wouldn't work as for root we are transferring instead of burning, so need to have that part separate

File: pallet/xrpl-bridge/src/lib.rs (hunk 2 above)
Comment (JasonTulp) [reply pallet/xrpl-bridge/src/lib.rs:1164]: Good idea, I'll make a ticket and we can tackle this after

File: pallet/xrpl-bridge/src/lib.rs (hunk 3 above)
Comment (JasonTulp) [reply pallet/xrpl-bridge/src/lib.rs:1316]: I believe asset_id is included in withdraw_tx

File: pallet/xrpl-bridge/src/lib.rs (hunk 4 above)
Comment (JasonTulp) [reply pallet/xrpl-bridge/src/lib.rs:55]: Done, I miss nightly

Comment (JasonTulp) [reply pallet/xrpl-bridge/src/lib.rs:55]: https://github.com/futureversecom/trn-seed/pull/872/commits/97201efd3aebfa9bd6eb35473870a2f6bc2dc335
